build.roc

```diff
@@ -3,81 +3,54 @@ app [main] {
 }
 
 import cli.Cmd
+import cli.Stdout
 
 main =
+    buildForSurgicalLinker!
+    buildForLegacyLinker!
 
-    # get the native target
-    native = getNativeTarget!
+exampleFile = "examples/hello-world.roc"
 
-    # build the target
-    buildGoTarget! { target: native, hostDir: "host", platformDir: "platform" }
+buildForSurgicalLinker : Task {} _
+buildForSurgicalLinker =
+    buildLibappSo!
+    buildDynhost!
+    preprocess!
 
-buildGoTarget : { target : RocTarget, hostDir : Str, platformDir : Str } -> Task {} _
-buildGoTarget = \{ target, hostDir, platformDir } ->
-
-    (goos, goarch, prebuiltBinary) =
-        when target is
-            MacosArm64 -> ("darwin", "arm64", "macos-arm64.a")
-            MacosX64 -> ("darwin", "amd64", "macos-x64")
-            LinuxArm64 -> ("linux", "arm64", "linux-arm64.a")
-            LinuxX64 -> ("linux", "amd64", "linux-x64.a")
-            WindowsArm64 -> ("windows", "arm64", "windows-arm64.a")
-            WindowsX64 -> ("windows", "amd64", "windows-x64")
+buildLibappSo =
+    Cmd.exec! "roc" ("build --lib $(exampleFile) --output host/libapp.so" |> Str.split " ")
 
+buildDynhost =
     Cmd.new "go"
-        |> Cmd.envs [("GOOS", goos), ("GOARCH", goarch), ("CC", "zig cc")]
-        |> Cmd.args ["build", "-C", hostDir, "-buildmode=c-archive", "-o", "libhost.a"]
+        |> Cmd.args ("build -C host -buildmode pie -o ../platform/dynhost" |> Str.split " ")
+        |> Cmd.envs [("GOOS", "linux"), ("GOARCH", "amd64"), ("CC", "zig cc")]
```

Comment:
Is this hardcoded for linux?

Reply:
This is only for the surgical linker. Since the surgical linker does only work on linux, I hardcoded it.